commons-vfs2/src/main/java/org/apache/commons/vfs2/impl/FileTypeMap.java

```diff
@@ -57,7 +58,7 @@ public String getScheme(final FileObject file) throws FileSystemException {
         // Check the file's mime type for a match
         final FileContent content = file.getContent();
         final String mimeType = content.getContentInfo().getContentType();
-        if (mimeType != null) {
+        if (mimeType != null && !mimeType.equals(MIME_TYPE_OCTET_STREAM)) {
```

Comment:
This does not feel right. One the one hand we tell caller to call `addMimeType(String, String)` but then we ignore that entry for one specific value. That does not make sense. I'd like to hear from others who might be more familiar with this part of the code.

Reply:
Not familiar with `mimeType` eithor, seems it's used in Internet media.

According to the [mimeType table](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types) the `.gz` should be regarded as `application/gzip`, but we got a `application/octet-stream` here. Actually I diged a little here and found it's caused in `sun.net.www.MimeTable` and will treat extensions `.saveme, .dump, .hqx, .arc, .obj, .lib, .bin, .exe, .gz` as `application/octet-stream`.Not sure if this is a potential problem of `sun.net.www.MimeTable` or not. 

This is just a workaround for it.

Reply:
Actually `application/octet-stream` is the default value for extensions not in the [mime type table](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types). I'm just adding one more extension check using `VFS's extensionMap` if we encountered a `application/octet-stream` here.